DEVELOPMENT_CONTRIBUTION.md

```diff
@@ -0,0 +1,159 @@
+# Contribution process for developers
+If you plan on contributing code to Clarity, please follow this step-by-step process in order to reduce the risk of major changes being requested when you submit your pull request. We'll work with you on every step and try to be as responsive as possible.
+
+## I. Proposal template
+
+Before you start coding anything, please fill out the following proposal template in as much detail as you can. The more complete your details the better, but not all questions will apply for every change.
+As you fill this out, please make sure to follow [our guidelines](/CODING_GUIDELINES.md#public-api) for the public API of components.
+```markdown
+## Summary
+
+Describe the change or feature in detail. Try to answer the following questions.
+
+* What is the change?
+* What is a use case for this change?
+* Why should it go in Clarity?
+* Does this change impact existing behaviors? If so how?
+* If this change introduces a new behavior, is this behavior accessible?
+
+
+## Examples
+
+_If possible, show examples of the change. You may create one yourself, or link to external sites that have the idea. It can also be very useful to prototype the idea in isolation outside of Clarity with a Plunkr or Stackblitz example._
+
+
+## API
+
+_Describe the intended API for the feature you want to add. This would include:_
+
+* CSS classes and DOM structure for pure static UI contribution, and inputs/outputs, components, directives, services, or anything that is exported publicly for Angular contributions.
+* Examples of code snippets using this new feature.
+* Note very clearly if anything **might** be a breaking change.
+
+_In the case of bug fixes or internal changes, there will most likely be no API changes._
+
+
+## Implementation Plan
+
+_Describe how you plan to implement the feature, answering questions among the following or anything else you deem relevant._
+
+* What parts of the code are affected?
+* Will you introduce new services, components or directives?
+* Can you describe the basic flow of information between classes?
+* Does this introduce asynchronous behaviors?
+* Will you need to refactor existing Clarity code?
+* Will reasonable performance require optimizations?
+* Will it need to access native elements (and be incompatible with server-side rendering)?
+* ...
+
+
+## Conclusion
+
+_Describe how long you expect it to take to implement, what help you might need, and any other details that might be helpful. Don't worry, this is obviously non-contractual. 😛_
+```
+
+Once it's ready, post it either on the original GitHub issue for bug fixes or already designed components, or in a new issue otherwise. If the original issue hasn't been updated in a while, please ping @youdz, @gnomeontherun or @adityarb88 on the issue to ensure we will start the discussion as soon as possible.
```

Comment:
If it matters, `@youdz` type references don't get linked to users. If you want to be fancy make them links to our pages.

Reply:
I'm ok with it not being a link here. People might read this in raw markup after they pull it locally, having these be links would just obfuscate them.